modules/sfp_c99.py

```diff
@@ -0,0 +1,624 @@
+            ip = domainHistoryElem.get("ip_address")
+
+            if ip is not None:
+                evt = SpiderFootEvent(
+                    "IP_ADDRESS",
+                    f"IP: {ip}, Date: {date}",
+                    self.__name__,
+                    event,
+                )
+                self.notifyListeners(evt)
+
+    def emitIPToSkypeData(self, data, event):
+        self.emitRawRirData(data, event)
+
+        skype = data.get("skype")
+
+        if skype is not None:
+            evt = SpiderFootEvent(
+                "ACCOUNT_EXTERNAL_OWNED",
+                skype,
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+            evt = SpiderFootEvent(
+                "USERNAME",
+                skype,
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+    def emitIPToDomainsData(self, data, event):
+        self.emitRawRirData(data, event)
+
+        domains = data.get("domains")
+
+        if isinstance(domains, list):
+            for domain in domains:
+                if self.checkForStop():
+                    return None
+
+                if self.opts["verify"] and not self.sf.resolveHost(domain):
+                    self.sf.debug(
+                        f"Host {domain} could not be resolved for {event.data}"
+                    )
+                    evt = SpiderFootEvent(
+                        "INTERNET_NAME_UNRESOLVED",
+                        domain,
+                        self.__name__,
+                        event,
+                    )
+                    self.notifyListeners(evt)
+                else:
+                    evt = SpiderFootEvent(
+                        "INTERNET_NAME",
+                        domain,
+                        self.__name__,
+                        event,
+                    )
+                    self.notifyListeners(evt)
+
+    def emitProxyDetectionData(self, data, event):
+        self.emitRawRirData(data, event)
+        isProxy = data.get("proxy")
+
+        if isProxy is not None:
+            evt = SpiderFootEvent(
+                "WEBSERVER_TECHNOLOGY",
+                f"Server is proxy: {isProxy}",
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+    def emitGeoIPData(self, data, event):
+        self.emitRawRirData(data, event)
+
+        hostName = data.get("hostname")
+        if (
+            hostName is not None
+            and self.opts["verify"]
+            and not self.sf.resolveHost(hostName)
+        ):
+            self.sf.debug(f"Host {hostName} could not be resolved for {event.data}")
+            evt = SpiderFootEvent(
+                "INTERNET_NAME_UNRESOLVED",
+                hostName,
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+        else:
```

Comment:
Based on the `if` clause, this will be executed when `hostName` is None, which I don't think is intended?